main/src/ca/uwaterloo/flix/language/errors/EntryPointError.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 Matthew Lutze
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.errors
+
+import ca.uwaterloo.flix.language.CompilationMessage
+import ca.uwaterloo.flix.language.ast.{SourceLocation, Symbol, Type}
+import ca.uwaterloo.flix.language.fmt.{Audience, FormatType}
+import ca.uwaterloo.flix.util.Formatter
+
+sealed trait EntryPointError extends CompilationMessage {
+  val kind: String = "Entry Point Error"
+}
+
+object EntryPointError {
+
+  private implicit val audience: Audience = Audience.External
+
+  /**
+    * Error indicating an unexpected argument type to an entry point function.
+    *
+    * @param sym the entry point function.
+    * @param tpe the argument type.
+    * @param loc the location where the error occurred.
+    */
+  case class UnexpectedEntryPointArg(sym: Symbol.DefnSym, tpe: Type, loc: SourceLocation) extends EntryPointError {
```

Comment:
I think internally we use the word "Illegal" (right?) but then in the messages we tend to use soft languages, e.g. "
unexpected"? At least that is what I think. Maybe add a line to `STYLE.md` if you agree :)

Reply:
I think the point is that the class names should be clear, but their text may change over time.